### PR TITLE
RR-647 - Add inductionDto and associated code

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR adds the `inductionDto` and associated code (enums, mapper & service) that we will need for migrating the CIAG UI screens to PLP UI
Pretty much all of this has been copied verbatim from the code we added in CIAG UI (with a few minor tweaks), so should look pretty familiar

### CODE REVIEW

File: server/@types/dto/index.d.ts [hunk 1]
```diff
@@ -30,3 +30,163 @@ declare module 'dto' {
     sequenceNumber: number
   }
 }
+
+declare module 'inductionDto' {
```
Comment (nathanrussell-moj-digital) [inline server/@types/dto/index.d.ts:34]: Copied verbatim from CIAG UI, just wrapped in a different module name

File: server/data/mappers/inductionDtoMapper.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import type { InductionResponse } from 'educationAndWorkPlanApiClient'
+import type { InductionDto } from 'inductionDto'
+
+const toInductionDto = (inductionResponse: InductionResponse): InductionDto => {
```
Comment (nathanrussell-moj-digital) [inline server/data/mappers/inductionDtoMapper.ts:4]: This mapper, and it's test, copied verbatim from CIAG UI

File: server/services/inductionService.ts [hunk 3]
```diff
@@ -19,18 +21,28 @@ export default class InductionService {
...
     try {
-      const induction = await this.getInduction(prisonNumber, token)
+      const induction = await this.retrieveInduction(prisonNumber, token)
       return toEducationAndTraining(induction)
     } catch (error) {
       return { problemRetrievingData: true } as EducationAndTraining
     }
   }
 
+  async getInduction(prisonNumber: string, token: string): Promise<InductionDto> {
```
Comment (nathanrussell-moj-digital) [inline server/services/inductionService.ts:31]: This is about the only part that has not been copied from CIAG UI (though it is based on their version)
We will need a service method to retrieve an Induction and map it into an `InductionDto`, and we will need it for pretty much every story that we are about to get into re: Updating the Induction; so it made sense to add it now.

We previously has a private method called `getInduction`, so I renamed that one to `retrieveInduction`, and then wrote this version of `getInduction`

File: server/testsupport/inductionDtoTestDataBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
+import type { InductionDto } from 'inductionDto'
```
Comment (nathanrussell-moj-digital) [inline server/testsupport/inductionDtoTestDataBuilder.ts:1]: Copied from CIAG UI

File: server/services/inductionService.ts (hunk 3 above)
Comment (mattwills-moj-digital) [reply server/services/inductionService.ts:31]: I found this a bit confusing initially. How about putting the error handling in the client and then I think you can get rid of `retrieveInduction()`?

**Review by mattwills-moj-digital — APPROVED** [review]
LGTM

**Pull request merged**